packages/boilerplate-generator/generator.js

```diff
@@ -23,13 +26,27 @@ export class Boilerplate {
   // purpose is to allow you to specify data that you might not know at
   // the time that you construct the Boilerplate object. (e.g. it is used
   // by 'webapp' to specify data that is only known at request-time).
+  // this returns three writeable objects:
+  // - a head that is a string for immediate flushing
+  // - a stream of the body
+  // - a closing body and scripts to flush and end the req
   toHTML(extraData) {
-    if (!this.baseData || !this.template) {
+    if (!this.baseData || !this.headTemplate || !this.closeTemplate) {
       throw new Error('Boilerplate did not instantiate correctly.');
     }
 
-    return  "<!DOCTYPE html>\n" +
-      this.template({ ...this.baseData, ...extraData });
+    const data = {...this.baseData, ...extraData};
+    const start = "<!DOCTYPE html>\n" + this.headTemplate(data);
+
+    const { body, dynamicBody } = data;
+    const stream = combine()
+      .append(body || "")
+      .append(dynamicBody || "")
+      .append(null);
+
+    const end = this.closeTemplate(data);
+
+    return { start, stream, end }
```

Comment:
Can't we just include the `start` and `end` chunks in the stream, so we don't have to return multiple things here?

Reply:
haha duh, wow should've done that from the start (see what I did there 😉)